### PR TITLE
`Paywalls`: new `PaywallActivityLauncher.launchIfNeeded` methods

### PR DESCRIPTION
**Pull request by NachoSoto**
This mimics the equivalent API in `PaywallDialogOptions` (and like in iOS).

### CODE REVIEW

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/helpers/HelperFunctions.kt [hunk 1]
```diff
@@ -3,7 +3,33 @@ package com.revenuecat.purchases.ui.revenuecatui.helpers
...
+    }
+
+    return shouldDisplayDialog
+}
+
+internal fun shouldDisplayBlockForEntitlementIdentifier(entitlement: String): ((CustomerInfo) -> Boolean) {
+    return { customerInfo ->
+        customerInfo.entitlements[entitlement]?.isActive != true
+    }
+}
```
Comment (NachoSoto) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/helpers/HelperFunctions.kt:35]: I've extracted these to share this code.

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/activity/PaywallActivityLauncher.kt [hunk 2]
```diff
@@ -41,4 +44,33 @@ class PaywallActivityLauncher {
     fun launch(offering: Offering? = null) {
         activityResultLauncher.launch(PaywallActivityArgs(offeringId = offering?.identifier))
     }
+
+    /**
+     * Launch the paywall activity if the current user does not have [requiredEntitlementIdentifier] active.
+     * @param offering The offering to be shown in the paywall. If null, the current offering will be shown.
+     */
+    @JvmOverloads
+    suspend fun launchIfNeeded(
```
Comment (NachoSoto) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/activity/PaywallActivityLauncher.kt:53]: Not sure if this is the best way to do this (exposing a `suspend` method). Feedback welcome.

Comment (tonidero) [reply ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/activity/PaywallActivityLauncher.kt:53]: Hmm well, I imagine many people using activities/fragments are still not using coroutines, so offering a wrapper would be better... However, we would only be able to launch this coroutine from a global scope unless we ask developers to give us the launcher activity/fragment/view model to act as the scope...

I see 2 options here:
- Store a weak reference to the activity/fragment on the constructor so we can use that to launch the coroutine in that scope, and offer a non-coroutine version of these methods.
- Have a non-coroutine version of `shouldDisplayPaywall` that returns a callback so we can support this use case. We can wrap that callback in a coroutine for the current version, same way we do in `CoroutinesExtensionsCommon`.

Of those 2, I think the second might be the cleanest, but the first one is probably ok.
Any other ideas @vegaro ?

Comment (NachoSoto) [reply ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/activity/PaywallActivityLauncher.kt]: Thanks! Just did option 2 :)

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/activity/PaywallActivityLauncher.kt [hunk 3]
```diff
@@ -41,4 +44,35 @@ class PaywallActivityLauncher {
     fun launch(offering: Offering? = null) {
         activityResultLauncher.launch(PaywallActivityArgs(offeringId = offering?.identifier))
     }
+
+    /**
+     * Launch the paywall activity if the current user does not have [requiredEntitlementIdentifier] active.
+     * @param offering The offering to be shown in the paywall. If null, the current offering will be shown.
```
Comment (tonidero) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/activity/PaywallActivityLauncher.kt:50]: We should add the docs for the `requiredEntitlementIdentifier` parameter.

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/activity/PaywallActivityLauncher.kt [hunk 4]
```diff
@@ -41,4 +44,35 @@ class PaywallActivityLauncher {
...
+    ) {
+        launchIfNeeded(
+            offering = offering,
+            shouldDisplayBlock = shouldDisplayBlockForEntitlementIdentifier(requiredEntitlementIdentifier),
+        )
+    }
+
+    /**
+     * Launch the paywall activity based on whether the result of [shouldDisplayBlock] is true.
+     * @param offering The offering to be shown in the paywall. If null, the current offering will be shown.
```
Comment (tonidero) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/activity/PaywallActivityLauncher.kt:65]: Same here about the docs for the new parameter.

**Review by tonidero — APPROVED** [review]
Looks great!

**Pull request merged**